tests/PHPStan/Analyser/nsrt/array-find-key.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace {
+
+	function array_find_key(array $array, callable $callback)
+	{
+		foreach ($array as $key => $value) {
+			if ($callback($value, $key)) {
+				return $key;
+			}
+		}
+
+		return null;
+	}
+
+}
+
+namespace ArrayFindKey
+{
+
+	use function PHPStan\Testing\assertType;
+
+	/**
+	 * @param array<mixed> $array
+	 */
+	function testMixed(array $array, callable $callback): void
+	{
+		assertType('int|string|null', array_find_key($array, $callback));
+		assertType('int|string|null', array_find_key($array, 'is_int'));
```

Comment:
You should test on non-empty-array<int> and it should gives int|string

Reply:
It seems like it can be improved at the same time as `array_filter()`.

https://phpstan.org/r/ab62bff1-d485-46ae-9dfa-1931ce320a25

Reply:
We can add a test case, but if you pass a `non-empty-array<mixed>` the function may return any `int|string|null`.

https://3v4l.org/Biok8

Reply:
Indeed, my bad.